### PR TITLE
Collection factories tests

### PR DESCRIPTION
**Pull request by butschster**

### CODE REVIEW

File: src/FactoryInterface.php [hunk 1]
```diff
@@ -84,6 +84,9 @@ public function relation(
         string $relation
     ): RelationInterface;
 
+    /**
+     * Add default classes for resolve
```
Comment (roxblnfk) [inline src/FactoryInterface.php]:
```suggestion
     * Add default classes for producing
```

File: tests/ORM/Unit/Collection/BaseTest.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                [
+                    'foo' => 'bar',
+                    'baz' => 'bar',
+                ],
+            ],
+            'generator' => [
+                $this->generatorArray(),
+            ],
+            'traversable' => [
+                new class () implements \Iterator {
```
Comment (roxblnfk) [inline tests/ORM/Unit/Collection/BaseTest.php]: Why didn't you use the `\ArrayIterator`?

Comment (butschster) [reply tests/ORM/Unit/Collection/BaseTest.php]: I'm not so smart like you :(((

Comment (roxblnfk) [reply tests/ORM/Unit/Collection/BaseTest.php]: You're better than you think.

**Review by roxblnfk — APPROVED** [review]

**Pull request merged**